src/aarch64/ffi.c

```diff
@@ -700,6 +705,10 @@ ffi_call_int (ffi_cif *cif, void (*fn)(void), void *orig_rvalue,
   void *stack, *rvalue;
   struct arg_state state;
   size_t stack_bytes, rtype_size, rsize;
+#ifdef __CHERI_PURE_CAPABILITY__
+  size_t frame_padding, rvalue_padding, stack_padding;
+  uintptr_t cursor;
```

Comment:
Use `char *` not `uintptr_t` for pointer arithmetic

Reply:
I've applied this suggestion. Could you tell why it's preferred over `uintptr_t`? It seemed more natural to me to use a data type that is not specific to a data type of referenced data (like `char` in this case).

Reply:
My basic philosophy is "don't use uintptr_t unless you need its added functionality". It's mostly about using the right type for the job; uintptr_t is a big hammer that you don't always need, and can have compiler optimisation implications, though that's not hugely important here. Note that char is special in C and can be used to alias any object.